core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/DedupAggregator.java

```diff
@@ -186,19 +187,30 @@ private void validateSchema(Schema inputSchema, List<String> uniqueFields,
       Schema.Field field = inputSchema.getField(uniqueField);
       if (field == null) {
         collector.addFailure(String.format("Field '%s' does not exist in the input schema", uniqueField), "")
-          .withConfigElement("uniqueFields", uniqueField);
+                .withConfigElement("uniqueFields", uniqueField);
       }
     }
 
     if (function != null) {
       Schema.Field field = inputSchema.getField(function.getField());
       if (field == null) {
         collector.addFailure(String.format("Invalid filter %s(%s): Field '%s' does not exist in input schema ",
-                                           function.getFunction(), function.getField(), function.getField()),
-                             null)
-          .withConfigProperty("filterOperation");
+                                function.getFunction(), function.getField(), function.getField()),
+                        null)
+                .withConfigProperty("filterOperation");
       }
-    }
+      List<Schema.Type> numericSchemaTypes = new ArrayList(Arrays.asList(Schema.Type.INT, Schema.Type.LONG,
+              Schema.Type.FLOAT, Schema.Type.DOUBLE));
+      Schema.Type fieldType = field.getSchema().isNullable() ?
+              field.getSchema().getNonNullable().getType() : field.getSchema().getType();
+      if ((function.getFunction().equals(DedupConfig.Function.MAX) ||
+              function.getFunction().equals(DedupConfig.Function.MIN)) && !numericSchemaTypes.contains(fieldType)) {
+        collector.addFailure(String.format("Unsupported filter operation %s(%s): Field '%s' is non numeric ",
```

Comment:
nit: arrange this so the max or min check is in the same line for better legibility:

```
      if ((function.getFunction().equals(DedupConfig.Function.MAX) || function.getFunction().equals(DedupConfig.Function.MIN)) 
        && !numericSchemaTypes.contains(fieldType)) {
 ```